stick to new dash manifest uri after http 302 redirection

davibe opened the pull request:
Related to discussion in #6907

davibe committed:
stick to new dash manifest uri after http 302 redirection

ojw28 commented on library/dash/src/main/java/com/google/android/exoplayer2/source/dash/DashMediaSource.java:
```diff
@@ -791,6 +791,18 @@ protected void releaseSourceInternal() {
     manifestLoadStartTimestampMs = elapsedRealtimeMs - loadDurationMs;
     manifestLoadEndTimestampMs = elapsedRealtimeMs;
+    boolean isRedirect = !loadable.getUri().equals(this.manifestUri);
+    if (isRedirect) {
```
The problem with having this as a separate block is that L811 will always evaluate to false if a redirect has occurred, which means that the location element wont be given priority. I think you need a single block similar to the suggested code snippet I put on the previous pull request.

davibe replied:
Right.

davibe committed:
DASHMediaSource: ensure MPD.Location is applied in also case of http redirect

davibe committed:
DASHMediaSource: avoid potentially locking twice

It many happen when changing manifest uri because of http 302 and MPD.Location present.
